feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-base.component.ts

```diff
@@ -358,4 +381,74 @@ export class ConfiguratorAttributeBaseComponent {
       !this.isReadOnly(attribute)
     );
   }
+
+  /**
+   * Creates a text describing the current attribute that can be used as ARIA label.
+   * Includes price information. If a total price is available this price will be used,
+   * otherwise it falls back to the value price, or if no price is available,
+   * no price information will be included in the text.
+   *
+   * @param attribute the attribute
+   * @param value the value
+   * @param considerSelectionState = false
+ - optional, depending on the underlying UI control the screen
+   * might announce the selection state on its own, so it is not always desired to include it here.
+   * @returns translated text
+   */
+  protected getAriaLabelGeneric(
+    attribute: Configurator.Attribute,
+    value: Configurator.Value,
+    considerSelectionState = false
+  ): string {
+    value =
+      this.configuratorDeltaRenderingService?.mergePriceIntoValue(value) ??
+      value;
+    const params: { value?: string; attribute?: string; price?: string } = {
+      value: value.valueDisplay,
+      attribute: attribute.label,
+    };
+
+    const includedSelected = considerSelectionState && value.selected;
+    let key = includedSelected
+      ? 'configurator.a11y.selectedValueOfAttributeFullWithPrice'
+      : this.getAriaLabelForValueWithPrice(this.isReadOnly(attribute));
+    if (value.valuePriceTotal && value.valuePriceTotal?.value !== 0) {
+      params.price = value.valuePriceTotal.formattedValue;
+    } else if (value.valuePrice && value.valuePrice?.value !== 0) {
+      params.price = value.valuePrice.formattedValue;
+    } else {
+      key = includedSelected
+        ? 'configurator.a11y.selectedValueOfAttributeFull'
+        : this.getAriaLabelForValue(this.isReadOnly(attribute));
+    }
+
+    let ariaLabel = '';
+    this.translation
+      .translate(key, params)
+      .pipe(take(1))
+      .subscribe((text) => (ariaLabel = text));
+
+    return ariaLabel;
+  }
+
+  /**
+   * Extract corresponding value price formula parameters.
+   * For all non-single selection types types the complete price formula should be displayed at the value level.
+   *
+   * @param {Configurator.Value} value - Configurator value
```

Comment:
@Uli-Tiger, could you, please, remove  curly braces for a parameter and a return value with their types, for instance:
> ~~{Configurator.Value}~~